README.md

```diff
@@ -0,0 +1,157 @@
+    Mopsos-->>-Argo CD: 200 OK
+    Mopsos-)Database: upsert info into records table
+    Note right of Database: The database can <br/>be queried directly <br/>or connected to <br/>dashboarding tools <br/>like metabase <br/>or grafana.
+```
+
+## Usage
+
+```console
+Mopsos receives events and stores them in a database for later analysis.
+
+Usage:
+  mopsos [flags]
+
+Flags:
+      --db-dsn string           Database DSN (default "file::memory:?cache=shared")
+      --db-migrate              Migrate database schema on startup (default true)
+      --db-provider string      Database provider, either 'sqlite' or 'postgres' (default "sqlite")
+      --debug                   Enable debug mode
+  -h, --help                    help for mopsos
+      --http-listener string    HTTP listener (default ":8080")
+      --otel                    Enable OpenTelemetry tracing
+      --otel-collector string   Endpoint for OpenTelemetry Collector. On a local cluster the collector should be accessible through a NodePort service at the localhost:30078 endpoint. Otherwise replace localhost with the collector endpoint. (default "localhost:30079")
+      --verbose                 Enable verbose mode
+```
+
+## Deployment
+
+The recommended way to deploy Mopsos is using Helm:
+
+```bash
+helm repo add adfinis https://charts.adfinis.com
+helm install adfinis/mopsos
+```
+
+Please refer to the [`mopsos` Helm chart](https://github.com/adfinis-sygroup/helm-charts/tree/master/charts/back8sup)
+for further information.
+
+### Telemetry
+
+You can send telemetry data to an [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/getting-started/) instance.
+
+| environment variable | default | comment |
+| ---- | ---- | ---- |
+| `MOPSOS_OTEL` | `"false"` | set to `"true"` to enable sending traces to the collector |
+| `MOPSOS_OTEL_COLLECTOR` | `"localhost:30079"` | needs to point to a grpc otlp receiver |
+
+## Development
+
+### Testing
+
+You can run the included unittest as follows.
+
+```bash
+# run tests (./... recursivley scans the repo for _test.go files)
+go test ./...
+
+# generate and view test coverage
+go test -coverprofile=coverage.out ./...
+go tool cover -html=coverage.out
+```
+
+These and more tests are also run on each PR as defined in
+[.github/workflows/lint-and-test.yaml](./.github/workflows/lint-and-test.yaml)
+
+### Running
+
+You can start a local instance of Mopsos using go run:
+
+```bash
+go run ./...
+```
+
+Once Mopsos reports that it is running you can send it a sample event using `curl`:
+
+```bash
+cat > event.json <<'EOD'
+{
+    "specversion": "1.0",
+    "type": "cloud.adfinis.mopsos.updateRecord",
+    "datacontenttype": "application/json",
+    "data": {
+        "cluster_name": "kubernetes.svc.cluster.invalid",
+        "application_name": "mopsos",
+        "application_version": "0.0.0"
+    }
+}
+EOD
+
+curl -v \
+     -H 'Content-Type: application/cloudevents+json' \
```

Comment:
did we figure out yet how to allow also `application/json` directly?

Reply:
no, not yet. i kinda decided that there are more important things to do first

we can fix this once we run into issues with it but the mvp should work just fine with the mixed `application/cloudevents+json` data type

i made a note in our internal tracker so i don't forget